app/visualization/views.py

```diff
@@ -0,0 +1,61 @@
+from flask import render_template, flash, redirect, url_for, request, send_from_directory
+from flask_login import login_required, current_user
+from flask import current_app as app
+from werkzeug import secure_filename
+from . import visualization
+from .. import db
+from ..models import Resource
+from wcwave_adaptors import default_vw_client
+from wcwave_adaptors import make_fgdc_metadata, metadata_from_file
+
+import os, osr, util, numpy
+
+
+VW_CLIENT = default_vw_client()
+
+
+@visualization.route('/')
+@login_required
+def confirmVisFile():
+    """
+    This function is designed for users to choose
+    a file to visualize
+    """
+    #TODO: upload file part
+
+    #TODO: obtain units from csv file
+
+    #TODO
+    #choose a file stored in the server 
+    #grab the url of the csv file
+    
+    
+
+    #this part is just for test
+    #for csv only
+    item_nameList = []
+    #get the current app location
+    #app/visualization
+    APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+    download_dir = APP_ROOT + '/tempData/'
+    filename = 'BRW_HrlySummary_2014.csv'
+    #getItemName five variables: folder, file name, item in the csv,
+    #first row offset, second row offset,  coloumn offset
+    util.getItemName(download_dir,filename,item_nameList ,19,0,0)
+   
+    return render_template('visualization/index.html', item_nameList = item_nameList)
+
```

Comment:
Hi Matthew,

Sorry that I forgot to explain this part. This is for initial version now.
I will create an interface for users to choose a csv file and fill some
information, such as row offsets and column offsets. Then the users can
visualize the csv file. Now the csv visualization section just visualize
the sample csv file you gave me.

Best regards,
Rui

On Thu, Jul 2, 2015 at 3:11 PM, Matthew <notifications@github.com> wrote:

> In app/visualization/views.py[github.com]
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mtpain_vwplatform_pull_41-23discussion-5Fr33827942&d=BQMCaQ&c=jifKnBYnyVBhk1h9O3AIXsy5wsgdpA1H51b0r9C8Lig&r=SBIVeg5P8-0Twl1xcR2vfw&m=UwqDIC1ahMdKqUGZNhY5sK2rb-C4sF6SmnsF5-IxuUU&s=8BbPVGT4vVJL749WV9FM7afWHfSGM9AhpeAhXN_-_9s&e=>
> :
>
> > +
> > +
> > +    #this part is just for test
> > +    #for csv only
> > +    item_nameList = []
> > +    #get the current app location
> > +    #app/visualization
> > +    APP_ROOT = os.path.dirname(os.path.abspath(__file__))
> > +    download_dir = APP_ROOT + '/tempData/'
> > +    filename = 'BRW_HrlySummary_2014.csv'
> > +    #getItemName five variables: folder, file name, item in the csv,
> > +    #first row offset, second row offset,  coloumn offset
> > +    util.getItemName(download_dir,filename,item_nameList ,19,0,0)
> > +
> > +    return render_template('visualization/index.html', item_nameList = item_nameList)
> > +
>
> As a user, how can I use this? A more explanatory docstring would be
> helpful for relation(path)
>
> —
> Reply to this email directly or view it on GitHub[github.com]
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mtpain_vwplatform_pull_41_files-23r33827942&d=BQMCaQ&c=jifKnBYnyVBhk1h9O3AIXsy5wsgdpA1H51b0r9C8Lig&r=SBIVeg5P8-0Twl1xcR2vfw&m=UwqDIC1ahMdKqUGZNhY5sK2rb-C4sF6SmnsF5-IxuUU&s=wS7ijb3hBS6q1RlvTW5JmMXJ4VEcubBmGxwspssfca4&e=>
> .
>